pkg/util/push/otel.go

```diff
@@ -69,6 +70,22 @@ func OTLPHandler(
 		}
 
 		reader := http.MaxBytesReader(nil, r.Body, int64(maxRecvMsgSize))
+		// Handle compression.
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(reader)
+			if err != nil {
+				return nil, err
+			}
+			reader = gr
```

Comment:
Given that we use `io.ReadAll(reader)` few lines later, this opens possibility for small requests to blow up entire server.

Maybe instead of `http.MaxBytesReader(nil, r.Body, int64(maxRecvMsgSize))` we should use `LimitedReader` with a reasonable limit (but with error if request is too large, not just EOF).

Reply:
`http.MaxBytesReader` is exactly like `LimitedReader` but instead of EOF it returns `error: request too large`. I've wrapped the new reader around it as well.